ssl/internal.h

```diff
@@ -4064,14 +4065,19 @@ struct ssl_st {
   bssl::UniquePtr<SSL_SESSION> session;
 
   // client_cipher_suites contains cipher suites offered by the client during
-  // the handshake, with preference order maintained. This field is NOT
-  // serialized and is only populated if used in a server context.
+  // the handshake and that are supported by this library, with preference order
+  // maintained. This field is NOT serialized and is only populated if used in
+  // a server context.
   bssl::UniquePtr<STACK_OF(SSL_CIPHER)> client_cipher_suites;
 
-  // client_cipher_suites_arr is initialized when |SSL_client_hello_get0_ciphers|
-  // is called with a valid |out| param. It holds the cipher suites offered by the
-  // client from |client_cipher_suites| in an array.
-  bssl::Array<uint16_t> client_cipher_suites_arr;
+  // all_client_cipher_suites contains the raw bytes for cipher suites offered
+  // by the client during the handshake (including those not supported by this
+  // library), with preference order maintained. This field may contain
+  // cipher IDs that are invalid.
+  bssl::UniquePtr<char> all_client_cipher_suites;
+
+  // Field length of ciphers in client hello
```

Comment:
NP: it's worth calling out 2^16 is the maximum size allowed by the RFC https://datatracker.ietf.org/doc/html/rfc8446#section-4.1.2 and this is separate from the SSLBuffer 2^16 limit.